src/Message/ApplePayAuthorizeResponse.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Omnipay\Vindicia\Message;
+
+class ApplePayAuthorizeResponse extends Response
+{
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function isSuccessful()
+    {
+        $statusCode = $this->data['statusCode'];
+        return ($statusCode >= 200 && $statusCode < 300) || $statusCode == 304;
+    }
+
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function getMessage()
+    {
+        return $this->data['message'];
+    }
+
+    /**
+     * Is the response successful?
+     * Throws an exception if there's no code.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function getCode()
+    {
+        return $this->data['statusCode'];
+    }
+
+    /**
+     * Is the response a redirect?
+     * Throws an exception if there's no code.
+     * Successful PayPal purchase responses are always redirects.
+     *
+     * @return boolean
+     * @throws \Omnipay\Common\Exception\InvalidResponseException
+     */
+    public function isRedirect()
+    {
+        return $this->isSuccessful();
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getRedirectUrl()
+    {
+        $transaction = $this->getTransaction();
+        if (isset($transaction)) {
+            return $transaction->getPayPalRedirectUrl();
+        }
+        return null;
+    }
+
+    /**
+     * @return string|null
```

Comment:
could you add docs explaining what all of these things you can get mean? also, if they're not actually useful, you could just delete them

Reply:
Going to add this in the next coming commit.